include/soci/odbc/soci-odbc.h

```diff
@@ -33,6 +33,10 @@
 #include <sqlext.h> // ODBC
 #include <string.h> // strcpy()
 
+#ifndef SQL_SS_LENGTH_UNLIMITED
```

Comment:
Since this define is SQLNCLI driver-specific, it is not defined in `sql.h`, but in `sqlncli.h`, similarly to `SQL_SS_UDT` and others. Part of the ODBC mess.

In `sqlncli.h` it is defined as follows:
```
// SQL_SS_LENGTH_UNLIMITED is used to describe the max length of
// VARCHAR(max), VARBINARY(max), NVARCHAR(max), and XML columns
#define SQL_SS_LENGTH_UNLIMITED             0
```

The PR looks good to me. Thanks!